Source/UI/NewUI/BillingForm/View/BillingFormPhoneNumberText.swift

```diff
@@ -1,101 +1,31 @@
+        delegate = self
     }
     
-    required init?(coder: NSCoder) {
+    required init(coder aDecoder: NSCoder) {
         fatalError("init(coder:) has not been implemented")
     }
-    
-    private func setup() {
-        keyboardType = .phonePad
-        updateTextFieldFormat(textField: self)
-        addTarget(self, action: #selector(textFieldDidChange), for: UIControl.Event.editingChanged)
-    }
-
-    /// Called when the Billing form Phone number text field is changed.
-    @objc private func textFieldDidChange(textField: UITextField) {
-        updateTextFieldFormat(textField: textField)
-    }
-    
-    // TODO: Copied from old code. Needs to be refactored
-    /// Called when the Billing form Phone number text field is changed.
-    private func updateTextFieldFormat(textField: UITextField) {
-        var targetCursorPosition = 0
-        if let startPosition = textField.selectedTextRange?.start {
-            targetCursorPosition = textField.offset(from: textField.beginningOfDocument, to: startPosition)
-        }
-
-        let phoneNumber = textField.text!
-        let formatted = partialFormatter.formatPartial(phoneNumber)
-        textField.text = formatted
 
-        if var targetPosition = textField.position(from: textField.beginningOfDocument, offset: targetCursorPosition) {
-            if targetCursorPosition != 0 {
-                let lastChar = formatted[formatted.index(formatted.startIndex,
-                                                         offsetBy: targetCursorPosition - 1)]
-                if lastChar == " " && previousTextCount < formatted.count && phoneNumber != formatted {
-                    guard let aTargetPosition = textField.position(from: textField.beginningOfDocument,
-                                                                   offset: targetCursorPosition + 1) else {
-                                                                    return
-                    }
-                    targetPosition = aTargetPosition
-                }
-            }
-            if (previousFormat.filter {$0 == " "}.count != formatted.filter {$0 == " "}.count) &&
-                phoneNumber != formatted {
-                guard let aTargetPosition = textField.position(from: textField.beginningOfDocument,
-                                                               offset: targetCursorPosition + 1) else {
-                                                                return
-                }
-                targetPosition = aTargetPosition
-            }
-            textField.selectedTextRange = textField.textRange(from: targetPosition, to: targetPosition)
-        }
-        previousTextCount = formatted.count
-        previousFormat = formatted
+    override func textFieldDidEndEditing(_ textField: UITextField, reason: UITextField.DidEndEditingReason) {
+        super.textFieldDidEndEditing(textField, reason: reason)
+        phoneNumberTextDelegate?.phoneNumberIsUpdated(number: text)
```

Comment:
Where is this `text` coming from? I can't see any value from the lifecycle update actually being used, should `text` come from this in any way?

Reply:
`text` is from parent `PhoneNumberTextField`